### PR TITLE
.github: Add missing lifecyle tests in nightly

### PR DESCRIPTION
**Pull request by TeddyAndrieux**

### CODE REVIEW

File: .github/workflows/downgrade-test.yaml [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+          ARTIFACTS_URL: ${{ inputs.to-artifacts-url }}
+      - name: Retrieve target product.txt from artifacts
+        run: >
+          curl --fail -LO -u ${{ secrets.ARTIFACTS_USER }}:${{ secrets.ARTIFACTS_PASSWORD }}
+          ${{ inputs.to-artifacts-url }}/product.txt
+      - name: Save target version and git commit in environment
+        run: |
+          source product.txt
+          echo "target_version=$VERSION" >> $GITHUB_ENV
+          echo "target_git_commit=$GIT" >>$GITHUB_ENV
```
Comment (gdemonet) [inline .github/workflows/downgrade-test.yaml:120]: `$GIT` isn't a commit SHA1, but the output of `git describe`, I didn't know we could just check out this value (TIL)... But I guess you tested that :wink:

File: .github/workflows/lifecycle-dev.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        id: get-url
+        run: |
+          search_url="https://artifacts.scality.net/search/last_success/commit/github/scality/metalk8s/Pre-merge/$dev_commit"
+          artifacts_name=$(curl --fail "$search_url" -u "${{ secrets.ARTIFACTS_USER }}:${{ secrets.ARTIFACTS_PASSWORD }}")
+          if [ -n "$artifacts_name" ]; then
+            echo "artifacts_url=https://artifacts.scality.net/builds/$artifacts_name" >> $GITHUB_OUTPUT
+          else
+            echo "Did not find artifacts for branch ${{ inputs.dev-branch }} - commit $dev_commit." >&2
+            exit 1
+          fi
```
Comment (gdemonet) [inline .github/workflows/lifecycle-dev.yaml]: A simpler approach would be to checkout `inputs.dev-branch`, then use the `scality/action-artifacts` action to find the artifacts link

**Review by gdemonet — DISMISSED** [review]
LGTM, the suggestion really isn't necessary

File: .github/workflows/downgrade-test.yaml (hunk 1 above)
Comment (TeddyAndrieux) [reply .github/workflows/downgrade-test.yaml:120]: Indeed but it works :smiley:

File: .github/workflows/lifecycle-dev.yaml (hunk 2 above)
Comment (TeddyAndrieux) [reply .github/workflows/lifecycle-dev.yaml]: Right, I will change this

**Review by gdemonet — APPROVED** [review]

**Pull request merged**